lib/data-types.js

```diff
@@ -481,6 +534,33 @@ DATEONLY.prototype._stringify = function _stringify(date) {
 };
 
 /**
+ * A YEAR column.
+ *
+ *
+ * @property YEAR
+ */
+function YEAR() {
+  if (!(this instanceof YEAR)) return new YEAR();
+}
+util.inherits(YEAR, ABSTRACT);
+
+YEAR.prototype.key = YEAR.key = 'YEAR';
+YEAR.prototype.toSql = function() {
+  return 'YEAR';
+};
+
+YEAR.prototype._stringify = function _stringify(date) {
+  return date;
+};
+
+YEAR.prototype.validate = function(value){
+  if (!Validator.isInt(String(value) || String(value).length !== 4 || String(value).length !== 2)){
```

Comment:
The check for a length of 2 is because MySQL supports the two digit year format as well (e.g. 99 --> 1999, 17 --> 2017). Technically, MySQL supports one digit too, so maybe adding a check for a length of 1 would be good too?

http://dev.mysql.com/doc/refman/5.7/en/year.html